File: truthscroll-mvp/scripts/convert_docx.js
#!/usr/bin/env node
// Simple DOCX -> verse-level JSON converter using mammoth
// Usage: node scripts/convert_docx.js input.docx output.json

const fs = require('fs');
const path = require('path');
const mammoth = require('mammoth');

const REF_RE = /^(\d?\s?[A-Za-z]+(?:[\sA-Za-z\-'.]*?)?)\s+(\d+):(\d+)\b/;

async function extract(inputPath) {
  const result = await mammoth.extractRawText({ path: inputPath });
  const raw = result.value || '';
  // Split into paragraphs by blank lines (preserve ordering)
  const paragraphs = raw.split(/\n\s*\n|\r\n\s*\r\n|\r\s*\r/).map(s => s.trim()).filter(Boolean);

  const verses = [];
  let lastKey = null;
  for (const p of paragraphs) {
    const line = p.replace(/\r?\n/g, ' ').trim();
    const m = line.match(REF_RE);
    if (m) {
      const book = m[1].trim();
      const chapter = m[2];
      const verse = m[3];
      const text = line.slice(m[0].length).trim();
      const key = `${book} ${chapter}:${verse}`;
      verses.push({ book, chapter: Number(chapter), verse: Number(verse), text });
      lastKey = key;
    } else {
      // Attach to last verse if exists, otherwise create anonymous entry
      if (lastKey && verses.length) {
        verses[verses.length - 1].text += ' ' + line;
      } else {
        verses.push({ text: line });
      }
    }
  }

  // Optionally, collapse consecutive anonymous entries into one
  return verses;
}

async function main() {
  const args = process.argv.slice(2);
  if (args.length < 2) {
    console.error('Usage: node scripts/convert_docx.js input.docx output.json');
    process.exit(2);
  }
  const input = args[0];
  const out = args[1];
  if (!fs.existsSync(input)) {
    console.error('Input not found:', input);
    process.exit(2);
  }
  try {
    const items = await extract(input);
    fs.mkdirSync(path.dirname(out), { recursive: true });
    fs.writeFileSync(out, JSON.stringify(items, null, 2), 'utf-8');
    console.log(`Wrote ${items.length} items to ${out}`);
  } catch (err) {
    console.error('Error converting:', err);
    process.exit(1);
  }
}

if (require.main === module) main();

File: truthscroll-mvp/scripts/convert_docx.py
#!/usr/bin/env python3
"""Simple DOCX -> JSON extractor for TruthScroll.

Usage:
  python scripts/convert_docx.py /path/to/input.docx data/raw/output.json

This is a best-effort extractor that reads paragraphs and attempts to detect verse
references of the form "BookName 1:1". It outputs a JSON array of objects with
optional `reference` and `text` fields. You should review and normalize the output.

Requires: python-docx
  pip install python-docx
"""
import sys
import re
import json
from docx import Document

REF_RE = re.compile(r"^([A-Za-z]+)\s+(\d+):(\d+)\b")

def extract(docx_path):
    doc = Document(docx_path)
    out = []
    current_ref = None
    for p in doc.paragraphs:
        text = p.text.strip()
        if not text:
            continue
        m = REF_RE.match(text)
        if m:
            book = m.group(1)
            chap = m.group(2)
            verse = m.group(3)
            # remove the leading reference from text
            content = text[m.end():].strip()
            ref = f"{book} {chap}:{verse}"
            out.append({"reference": ref, "text": content})
            current_ref = ref
        else:
            # try to attach to current_ref if present
            if current_ref:
                out.append({"reference": current_ref, "text": text})
            else:
                out.append({"text": text})
    return out

def main():
    if len(sys.argv) < 3:
        print("Usage: convert_docx.py input.docx output.json")
        sys.exit(2)
    inp = sys.argv[1]
    outp = sys.argv[2]
    data = extract(inp)
    with open(outp, 'w', encoding='utf-8') as f:
        json.dump(data, f, ensure_ascii=False, indent=2)
    print(f"Wrote {len(data)} items to {outp}")

if __name__ == '__main__':
    main()
